### PR TITLE
feat: select option dynamic font

### PR DESCRIPTION
**Pull request by mertmit**
## Change Summary

Choose light/dark font based on select option color
+fixed some type errors for select cells

## Change type

- [x] feat: (new feature for the user, not a new feature for build script)

## Test/ Verification
image

### CODE REVIEW

File: packages/nc-gui/components/cell/MultiSelect.vue [hunk 1]
```diff
@@ -144,7 +146,11 @@ watch(isOpen, (n, _o) => {
   >
     <a-select-option v-for="op of options" :key="op.id" :value="op.title" @click.stop>
       <a-tag class="rounded-tag" :color="op.color">
-        <span class="text-slate-500" :class="{ 'text-sm': isKanban }">{{ op.title }}</span>
+        <span
+          :style="{ color: tinycolor.mostReadable(op.color || '#ccc', ['#64748b', '#f0f0f0']).toHex8String() }"
+          :class="{ 'text-sm': isKanban }"
+          >{{ op.title }}</span
+        >
```
Comment (bcakmakoglu) [inline packages/nc-gui/components/cell/MultiSelect.vue:153]: Can you fix the formatting here?
```html
<span
   :style="{ color: tinycolor.mostReadable(op.color || '#ccc', ['#64748b', '#f0f0f0']).toHex8String() }"
   :class="{ 'text-sm': isKanban }"
>
   {{ op.title }}
</span>
```

File: packages/nc-gui/components/cell/MultiSelect.vue [hunk 2]
```diff
@@ -153,12 +159,16 @@ watch(isOpen, (n, _o) => {
         v-if="options.find((el) => el.title === val)"
         class="rounded-tag"
         :style="{ display: 'flex', alignItems: 'center' }"
-        :color="options.find((el) => el.title === val).color"
+        :color="(options.find((el) => el.title === val) as SelectOptionType).color"
```
Comment (bcakmakoglu) [inline packages/nc-gui/components/cell/MultiSelect.vue:162]: Please don't cast to a type in the template, in VSCode that's okay but in Webstorm this breaks the whole file for me 😅
If you need to force a type, use a function that returns the correct type for you, that way I can have less of a headache in Webstorm xD

File: packages/nc-gui/components/cell/MultiSelect.vue [hunk 3]
```diff
@@ -153,12 +159,16 @@ watch(isOpen, (n, _o) => {
...
         :closable="active && (vModel.length > 1 || !column?.rqd)"
         :close-icon="h(MdiCloseCircle, { class: ['ms-close-icon'] })"
         @close="onClose"
       >
-        <span class="w-full text-slate-500" :class="{ 'text-sm': isKanban }">{{ val }}</span>
+        <span
+          :style="{ color: tinycolor.mostReadable((options.find((el) => el.title === val) as SelectOptionType).color || '#ccc', ['#64748b', '#f0f0f0']).toHex8String() }"
+          :class="{ 'text-sm': isKanban }"
+          >{{ val }}</span
+        >
```
Comment (bcakmakoglu) [inline packages/nc-gui/components/cell/MultiSelect.vue:171]: Formatting

File: packages/nc-gui/components/cell/SingleSelect.vue [hunk 4]
```diff
@@ -82,7 +83,11 @@ watch(isOpen, (n, _o) => {
   >
     <a-select-option v-for="op of options" :key="op.title" :value="op.title" @click.stop>
       <a-tag class="rounded-tag" :color="op.color">
-        <span class="text-slate-500" :class="{ 'text-sm': isKanban }">{{ op.title }}</span>
+        <span
+          :style="{ color: tinycolor.mostReadable(op.color || '#ccc', ['#64748b', '#f0f0f0']).toHex8String() }"
+          :class="{ 'text-sm': isKanban }"
+          >{{ op.title }}</span
+        >
```
Comment (bcakmakoglu) [inline packages/nc-gui/components/cell/SingleSelect.vue:90]: Formatting

**Review by bcakmakoglu — CHANGES_REQUESTED** [review]

File: packages/nc-gui/components/cell/MultiSelect.vue (hunk 2 above)
Comment (mertmit) [reply packages/nc-gui/components/cell/MultiSelect.vue:162]: oh okay let me see that

**Review by bcakmakoglu — APPROVED** [review]

**Pull request merged**